protocol/handshake_test.go

```diff
@@ -0,0 +1,68 @@
+package protocol
+
+import (
+	"bufio"
+	"bytes"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestHandshake_Do(t *testing.T) {
```

Comment:
what is `Do`? 
this name implies we have a method named `handshake.Do` according to [go naming convention](https://golang.org/pkg/testing/#hdr-Examples).
what do you think about simply:
```suggestion
func TestHandshake(t *testing.T) {
```

Reply:
sure, I forgot to rename it